### PR TITLE
Avoid failure while fetching package.json for rush repos

### PR DESCRIPTION
**Pull request by milind009**
# Issue
Currently, for a rush repo, if `package.json` does not exist at the specified root directory, dependabot-core raises error `Dependabot::DependencyFileNotFound` but for rush repos since `rush.json` already exists, it is not mandatory to have a `package.json` too.

# Solution
Instead of raising an error when package.json does not exist, added method to raise an error if no `manifest files(i.e rush.json, package.json)` exist else return an array of all the existing manifest files

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb [hunk 1]
```diff
@@ -135,7 +139,13 @@ def fetch_rush_configs
           fetch_file_if_present("common/config/pnpmfile-dependencies.json"),
           fetch_file_if_present("common/config/rush/common-versions.json"),
           fetch_file_if_present("common/config/rush/version-policies.json")
-        ].compact        
+        ].compact
+      end
+
+      def fetch_manifest_files
```
Comment (GiriB) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: would fetch_root_manifest be better name or something similar but with "root" in the name

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb [hunk 2]
```diff
@@ -22,11 +22,11 @@ class FileFetcher < Dependabot::FileFetchers::Base
       PATH_DEPENDENCY_CLEAN_REGEX = /^file:|^link:/.freeze
 
       def self.required_files_in?(filenames)
-        filenames.include?("package.json")
+        filenames.include?("package.json", "rush.json")
```
Comment (GiriB) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: Are we sure that rush.json should be a must? Are there legal cases where even rush.json is missing?

Comment (milind009) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: My bad. This change is not correct. Reverted it

File: npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb (hunk 1 above)
Comment (milind009) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb]: Makes sense. Updated

**Review by GiriB — APPROVED** [review]

**Pull request merged**